joplin/api/schema.py

```diff
@@ -340,22 +341,115 @@ class Meta:
         interfaces = [graphene.Node]
 
 
-class ProcessPageStepNode(DjangoObjectType):
-    class Meta:
-        model = ProcessPageStep
-        interfaces = [graphene.Node]
-
-
-class ProcessPageContactNode(DjangoObjectType):
-    class Meta:
-        model = ProcessPageContact
-        interfaces = [graphene.Node]
-
-
-class ProcessPageTopicNode(DjangoObjectType):
-    class Meta:
-        model = ProcessPageTopic
-        interfaces = [graphene.Node]
+class SiteStructure(graphene.ObjectType):
+    value = GenericScalar()
+    structure_json = JSONString()
+
+    # json isn't a great way to do this, we should
+    # figure out how to make it queryable
+    def resolve_structure_json(self, resolve_info, *args, **kwargs):
+        # our structure here can be id: page dict
+        site_structure = []
+        topic_collections = TopicCollectionPage.objects.all()
+        for topic_collection in topic_collections:
+            topic_collection_global_id = graphene.Node.to_global_id('TopicCollectionNode', topic_collection.id)
+            site_structure.append({'url' :f'/{topic_collection.theme.slug}/{topic_collection.slug}/', 'type': 'topic collection', 'id': topic_collection_global_id});
```

Comment:
for topic collections, the url is always created the same way (theme/tc) and there isn't any extra info we need that changes based on url/context